atomate/vasp/firetasks/surface_tasks.py

```diff
@@ -0,0 +1,174 @@
+
+
+"""
+Surface workflow firetasks.
+"""
+
+__author__ = "Richard Tran"
+__email__ = 'rit001@eng.ucsd.edu'
+
+
+@explicit_serialize
+class FacetFWsGeneratorTask(FiretaskBase):
+    """
+    Task for generating FWs for oriented unit cell calculations or slab
+    calculations. If the initial structure is an oriented unit cell,
+    generate slab fws, if its a conventional unit cell, generate oriented
+    unit cell and reconstruction calculations.
+    """
+
+    required_params = ['structure_type', "k_product", "vasp_cmd", "scratch_dir"]
+    optional_params = ["slab_gen_params", "mpid", "mmi", "db_file", "miller_index", "cwd"]
+
+    def run_task(self, fw_spec):
+        """
+        Generates a list of oriented unit cell FWs or
+        slab cell FWs depending on the structure type.
+
+        Args:
+            structure_type (str): Type of structure. Options are:
+                conventional_unit_cell, oriented_unit_cell and slab_cell
+            scratch_dir (str): - if specified, uses this directory as the root
+                scratch dir. Supports env_chk.
+            k_product (int): Default to 50, kpoint number * length for a & b
+                directions, also for c direction in bulk calculations.
+            db_file (str): FULL path to file containing the database credentials.
+                Supports env_chk.
+            vasp_cmd (str): Command to run vasp.
+            slab_gen_params (dict): Parameters for SlabGenerator or generate_all_slabs
+            mpid (str): Materials Project ID of the conventional unit cell.
+            mmi (int): Max Miller index.
+            miller_index ([h, k, l]): Miller index of plane parallel to
+                surface (and oriented unit cell).
+            cwd (str): Location of directory to operate the
+                workflow and store the final outputs
+            \*\*kwargs: Other kwargs that are passed to Firework.__init__.
+        """
+
+        # get the parameters for SlabGenerator if we are getting a
+        # list of Slab calculation FWs or generate_all_slabs if we
+        # are getting a list of oritend unit cell calculation FWs
+        if self.get("slab_gen_params", None):
+            slab_gen_params = self.get("slab_gen_params", None)
+        else:
+            slab_gen_params = {"min_slab_size": 10, "min_vacuum_size": 10,
+                               "symmetrize": True, "center_slab": True}
+            slab_gen_params["max_normal_search"] = self.get("mmi") if \
+                self.get("mmi") else max(self.get("miller_index"))
+            if self.get("structure_type") == "conventional_unit_cell":
+                slab_gen_params["include_reconstructions"] = True
+
+        FWs = []
+        if self.get("structure_type") == "conventional_unit_cell":
+            # Then we create a set of FWs for oriented_unit_cell
+            # calculations and reconstructed slabs
+            slab_gen_params["structure"] = Structure.from_file("CONTCAR.relax2.gz")
```

Comment:
The hard coding of CONTCAR.relax2.gz is not acceptable. Assumes:

-CONTCAR is in current dir
- was done with a 2-relaxation run
- is gzipped

This needs major change

Reply:
Note that:
- requiring CONTCAR to be in current dir is not so bad
- some of the gz stuff can be handled by monty zpath